src/users/users.controller.ts

```diff
@@ -96,25 +98,25 @@ export class UsersController {
       },
     },
   })
-  async getUser(@Param('email') email: string): Promise<UserDto> {
+  async getUser(@Param('query') query: string): Promise<UserDto> {
     try {
-      this.logger.debug(`Find a user by email: ${email}`);
-      return new UserDto(await this.usersService.findByEmail(email));
+      this.logger.debug(`Find a user by email/id: ${query}`);
+      return new UserDto(await this.usersService.findUser(query));
     } catch (err) {
       throw new HttpException(err.message, err.status);
     }
   }
 
-  @Get('/:id')
+  @Get('/full/:query')
```

Comment:
What does `full` mean?   The meaning and wording are too odd, it is not clear even to me.